AK/StringBase.cpp

```diff
@@ -79,6 +74,16 @@ bool StringBase::operator==(StringBase const& other) const
     return bytes() == other.bytes();
 }
 
+ErrorOr<u8*> StringBase::replace_with_uninitialized_buffer(size_t byte_count)
+{
+    if (byte_count <= max_short_string_byte_count)
+        return replace_with_uninitialized_short_string(byte_count);
+
+    u8* buffer = nullptr;
+    *this = StringBase { TRY(StringData::create_uninitialized(byte_count, buffer)) };
```

Comment:
This will explicitly slice the derived class. Can we just replace the relevant members instead?